csharp/src/Ice/Ice1NetworkSocket.cs

```diff
@@ -323,8 +335,25 @@ internal override ValueTask<SocketStream> SendInitializeFrameAsync(CancellationT
             }
             else
             {
-                return new ValueTask<SocketStream>(CreateStream(false));
+                return new ValueTask<SocketStream>(new Ice1NetworkSocketStream(AllocateId(false), this));
+            }
+        }
+
+        private long AllocateId(bool bidirectional)
```

Comment:
It's not obvious why we allocate unidirectional Socket IDs with ice1, since ice1 sends the same request ID (0) for all oneway requests.

Reply:
The Ice1 network socket behaves like other multi stream sockets and provide the same stream ID numbering scheme as Quic, the request ID is derived from the stream ID.

It's possible we could avoid assigning stream IDs to Ice1 network socket streams and instead directly use request IDs but I suspect this will be more complex and require callers to check if the stream is an Ice1 one or not.